mlflow/pyfunc/__init__.py

```diff
@@ -310,7 +311,7 @@ def _enforce_type(name, values: pandas.Series, t: DataType):
                 "Failed to convert column {0} from type {1} to {2}.".format(name, values.dtype, t)
             )
 
-    if values.dtype in (t.to_pandas(), t.to_numpy()):
+    if t.to_pandas() == values.dtype or t.to_numpy() == values.dtype:
```

Comment:
NB: comparison of pandas and numpy data type fails when numpy data type is on the left hand side of the comparison operator. It works, however, if pandas type is on the left hand side. That is because pandas is aware of numpy.

Reply:
that makes sense!